packages/design-system/src/components/contextMenu/styled.js

```diff
@@ -52,38 +51,7 @@ export const Shadow = styled.div`
 `;
 
 export function SmartPopover(props) {
-  const setRef = useCallback(
-    (node) => {
-      if (!node) {
-        return;
-      }
-
-      const boundingBox = node.getBoundingClientRect();
-      const max = {
-        x: window.innerWidth - boundingBox.width,
-        y: window.innerHeight - boundingBox.height,
-      };
-
-      // This is modeling the behavior of chrome:
-      // - menu becomes right justified if not enough space horizontally
-      // - menu sticks to bottom if not enough space vertically
-      const horizontalEdgeCondition = props.isRTL
-        ? boundingBox.x < 0
-        : max.x < boundingBox.x;
-      const horizontalEdgeTransform = props.isRTL
-        ? boundingBox.width
-        : -boundingBox.width;
-      const delta = {
-        x: horizontalEdgeCondition ? horizontalEdgeTransform : 0,
-        y: Math.min(0, max.y - boundingBox.y),
-      };
-      node.style.setProperty('--delta-x', delta.x);
-      node.style.setProperty('--delta-y', delta.y);
-    },
-    [props.isRTL]
-  );
-
-  return props.isOpen ? <Popover ref={setRef} {...props} /> : null;
+  return props.isOpen ? <Popover {...props} /> : null;
```

Comment:
This caused bug 1. and seems unnecessary, "Chrome behavior" is already done somewhere else (in `updatePosition`). From my testing removing it doesn't cause any issue elsewhere.

Reply:
Wondering if the `SmartPopover` should just be removed --- it's just a Popover with props now.